converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Modules/ServiceCollectionExtensions.cs

```diff
@@ -15,68 +16,78 @@ namespace Microsoft.Health.DicomCast.Core.Modules
 {
     public static class ServiceCollectionExtensions
     {
-        public static void AddNamedManagedIdentityCredentialProvider(this IServiceCollection serviceCollection, ManagedIdentityCredentialConfiguration managedIdentityCredentialConfiguration, string name)
+        public static void AddNamedManagedIdentityCredentialProvider(this IServiceCollection serviceCollection, IConfiguration managedIdentityCredentialConfiguration, string name)
         {
             EnsureArg.IsNotNull(serviceCollection, nameof(serviceCollection));
             EnsureArg.IsNotNull(managedIdentityCredentialConfiguration, nameof(managedIdentityCredentialConfiguration));
             EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
 
+            serviceCollection.Configure<ManagedIdentityCredentialConfiguration>(name, managedIdentityCredentialConfiguration);
+
             serviceCollection.Add(provider =>
                 {
-                    var options = Options.Create(managedIdentityCredentialConfiguration);
+                    IOptionsMonitor<ManagedIdentityCredentialConfiguration> options = provider.GetService<IOptionsMonitor<ManagedIdentityCredentialConfiguration>>();
```

Comment:
`Monitor` will change values mid-request, is that preferable to `Snapshot`'s behavior? I'm not familiar enough with the network patterns here to have strong opinion.

Reply:
It's definitely a bit odd because we're creating a `Singleton` service that relies on options where we have a monitor but are removing it. So if there are changes, this service won't see them. But I think Brandon is going to use `IOptionsMonitor` for these providers in a future PR. Then this custom service becomes moot.